crates/anvil/src/eth/backend/executor.rs

```diff
@@ -304,12 +310,19 @@ impl<DB: Db + ?Sized, V: TransactionValidator> Iterator for &mut TransactionExec
         let env = self.env_for(&transaction.pending_transaction);
 
         // check that we comply with the block's gas limit, if not disabled
-        let max_gas = self.gas_used.saturating_add(env.tx.base.gas_limit);
-        if !env.evm_env.cfg_env.disable_block_gas_limit && max_gas > env.evm_env.block_env.gas_limit
+        let max_block_gas = self.gas_used.saturating_add(env.tx.base.gas_limit);
+        if !env.evm_env.cfg_env.disable_block_gas_limit
+            && max_block_gas > env.evm_env.block_env.gas_limit
         {
-            return Some(TransactionExecutionOutcome::Exhausted(transaction));
+            return Some(TransactionExecutionOutcome::BlockGasExhausted(transaction));
         }
 
+        // TODO!
+        // check that we comply with the transaction's gas limit, if enabled
+        // if env.evm_env.cfg_env.tx_gas_limit_cap.is_none() {
+        // return Some(TransactionExecutionOutcome::TransactionGasExhausted(transaction));
+        // }
```

Comment:
To do: perform a validation check here, could possibly not be required as Revm will catch this itself